publish-package/README.md

```diff
@@ -0,0 +1,51 @@
+# Publish-PyPI-Package
+
+This Task publishes python packages to PyPI index using [Twine](https://pypi.org/project/twine/) utility module. It provides build system independent uploads of source and binary distribution artifacts for both new and existing projects.
+
+## Install the Task
+
+```
+kubectl apply -f https://raw.githubusercontent.com/tektoncd/catalog/master/publish-package/update-pypi.yaml
+```
+
+## Requirement
+
+Secret is needed with Twine credentials(TWINE_USERNAME:TWINE_PASSWORD) to be provided for python module publishing to PyPI index.
+
+- **username**: The username to authenticate to the repository (package index).
+- **password**: The password to authenticate to the repository (package index).
+
+```
+apiVersion: v1
+kind: Secret
+metadata:
+  name: pypi-secret
+type: kubernetes.io/basic-auth
+stringData:
+  username: foo
+  password: bar
+```
+
+## Resources
+
+- **source**: A `git`-type `PipelineResource` specifying the location of the source to build.
```

Comment:
I think we would want to use the `git-task.yaml` and workspaces instead of the `PipelineResources`, see for example the buildah task : 

https://github.com/tektoncd/catalog/blob/v1beta1/buildah/buildah.yaml#L28-L29
https://github.com/tektoncd/catalog/blob/v1beta1/buildah/buildah.yaml#L38

and the way we test it in there : 

https://github.com/tektoncd/catalog/blob/v1beta1/buildah/tests/pre-apply-task-hook.sh#L8
https://github.com/tektoncd/catalog/blob/v1beta1/buildah/tests/run.yaml#L6-L12

The README would need to be updated too : 

https://github.com/tektoncd/catalog/tree/v1beta1/buildah#workspaces